tools/clang/unittests/HLSLExec/LongVectors.cpp

```diff
@@ -616,12 +616,105 @@ DEFAULT_OP_2(OpType::Ldexp, (A * static_cast<T>(std::pow(2.0f, B))));
 // Bitwise
 //
 
+template <typename T> T Saturate(T A) {
+  if (A < static_cast<T>(0.0))
```

Comment:
nit:
```suggestion
  if (A < static_cast<T>(0.0f))
```
`f` is used below in the half impl